src/chargepoint/status/StatusManager.cpp

```diff
@@ -363,20 +363,43 @@ bool StatusManager::handleMessage(const ocpp::messages::ChangeAvailabilityReq& r
     unsigned int connector_id = request.connectorId;
     if (m_connectors.isValid(connector_id))
     {
-        // Notify request
-        response.status = m_events_handler.changeAvailabilityRequested(connector_id, request.type);
-        if (response.status == AvailabilityStatus::Accepted)
-        {
-            // Update status
-            ChargePointStatus status = ChargePointStatus::Unavailable;
-            if (request.type == AvailabilityType::Operative)
+        //In the case the ChangeAvailability.req contains ConnectorId = 0, the status change applies to the Charge Point and all Connectors.
+        if(connector_id == 0)
+        {            
+            for(unsigned int i=0;i <= m_connectors.getCount();i++)
             {
-                status = ChargePointStatus::Available;
+                response.status = m_events_handler.changeAvailabilityRequested(i, request.type);
```

Comment:
For connector 0, I would have done it differently : I would have notified the user application of a change availability request on connector 0 only. If the request has been accepted, I would have change the status of all the connectors.
By doing this, I would have only 1 status for my response to the Central System reflecting the overall change availability request status of the connectors.